include/wil/cppwinrt_helpers.h

```diff
@@ -212,5 +212,71 @@ namespace wil::details
     };
 }
 #endif // __WIL_CPPWINRT_MICROSOFT_UI_DISPATCHING_HELPERS
-
 /// @endcond
+
+#if defined(WINRT_Windows_Foundation_Collections_H) && !defined(__WIL_CPPWINRT_WINDOWS_FOUNDATION_COLLECTION_HELPERS)
+#define __WIL_CPPWINRT_WINDOWS_FOUNDATION_COLLECTION_HELPERS
+
+namespace wil
+{
+    /// @cond
+    namespace details
+    {
+        template<typename TData, typename TSrc> std::vector<TData> to_vector_impl(TSrc&& fetcher)
+        {
+            std::vector<TData> output;
+            const uint32_t chunkSize = 64;
+            while (true)
+            {
+                TData nextChunk[chunkSize]{ winrt::impl::empty_value<TData>() };

```

Comment:
I had a previous model that fetched data right into the vector, but that required an initial allocation on the storage even if the underlying IVector was zero-length.

This model causes a move from the local storage into an allocator, but since these are all scalar or pointer-sized, maybe that's OK?  I'm pretending that `vector<scalar>.insert` could have some optimization that blits from the input array to the output.

Not sure there's a good way around this.  (You can see the previous model in an earlier commit...)

Reply:
Using the "get everything in at most two calls" pattern now.